utils/extra.py

```diff
@@ -1,17 +1,20 @@
+from __future__ import annotations
+
 import io
 import os
 import pathlib
 import random
 import sys
 import zlib
-from typing import NamedTuple
+import typing
```

Comment:
from typing import Optional, Any, NamedTuple

Reply:
importing type also works

Reply:
I agree with Crunchy, the convention or at least what most people do is to import the individual elements from typing,
it is more readable and makes more sense.
especially since its only 3

Reply:
Good point.

Reply:
Done.